### PR TITLE
Time based animations

### PR DESCRIPTION
**Pull request by IvanSavenko**
Part of #1376
Adventure map will be done separately, at some point later. Hard to do anything with it without breaking at the moment.

- [x] Video playback speed now uses value supplied by decoder. Also fixes WoG/Tides of War main menu animation. 
(although at least ToF menu now plays notably *slower* than in 1.1 since video was recorded at 12 fps, but 1.1 played it at 16 fps)
- [x] Adventure map - map reveal via Eye of the Magi - confirmed to match H3
- [x] Adventure map - hourglass in infobox during enemy turn - fixed to match H3
- [x] Adventure map - new day animation - fixed to match H3
- [x] Town screen - buildings animation - fixed to match H3
- [x] Town screen - new building fade-in - fixed to match H3
- [x] Creature window - creature animation speed - fixed to match H3
- [x] Cursor - animation speed of spell selection cursor - fixed to match H3
- [x] Battles - all animations should match H3 
- [x] Battles - hero & hero flag animation

Minor todo - sometimes there is some sort of freeze for ~100-200ms when unit starts move. Can't remember it from develop. UPD: bug exists in develop

### CODE REVIEW

File: client/windows/CCastleInterface.h [hunk 1]
```diff
@@ -45,9 +45,11 @@ class CBuildingRect : public CShowableAnim
 	enum EBuildingCreationAnimationPhases : uint32_t
 	{
 		BUILDING_APPEAR_TIMEPOINT = 500, //500 msec building appears: 0->100% transparency
-		BUILDING_WHITE_BORDER_TIMEPOINT = 1000, //500 msec border glows from white to yellow
-		BUILDING_YELLOW_BORDER_TIMEPOINT = 1500, //500 msec border glows from yellow to normal
-		BUILD_ANIMATION_FINISHED_TIMEPOINT = 2500 //1000 msec delay, nothing happens
+		BUILDING_WHITE_BORDER_TIMEPOINT = 900, //400 msec border glows from white to yellow
+		BUILDING_YELLOW_BORDER_TIMEPOINT = 1100, //200 msec border glows from yellow to normal (dark orange)
```
Comment (dydzio0614) [inline client/windows/CCastleInterface.h:49]: BTW take into account hd mod or hota (forgot which one) speeds up the glow - I am unsure if you used base H3 as reference

Comment (dydzio0614) [reply client/windows/CCastleInterface.h:49]: also for me on linux H3 with HD seems to be missing stuff like black screen fade out when entering towns and battles etc - can test without HD later.

Comment (IvanSavenko) [reply client/windows/CCastleInterface.h:49]: And here I thought that I disabled all "enhancements" from HD Mod.
Yeah, I've noticed that this animation feels "wrong" but thought that this is just H3 behavior. 
Will recheck.

Comment (dydzio0614) [reply client/windows/CCastleInterface.h:49]: I would keep separate comment for HD mod timing - may be useful for future use

Comment (IvanSavenko) [reply client/windows/CCastleInterface.h:49]: Fixed. All other timings look OK, for battles will use vanilla game timings. Have even less trust in HD mod timings in battles

**Comment by IvanSavenko** [comment]
All animations I found other than those that are adventure map-related have now correct timings.
Should be ready for merge. Adventure map will be done later, need to look into & cleanup adventure map code first.

**Pull request merged**